src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go

```diff
@@ -12,58 +12,114 @@ import (
 	deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
 	sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
 	node_base "github.com/filecoin-project/specs/systems/filecoin_nodes/node_base"
+	actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
 	addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+	ai "github.com/filecoin-project/specs/systems/filecoin_vm/actor_interfaces"
+	msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
 	stateTree "github.com/filecoin-project/specs/systems/filecoin_vm/state_tree"
 	util "github.com/filecoin-project/specs/util"
 )
 
 type Serialization = util.Serialization
 
+// Note that implementations may choose to provide default generation methods for miners created
+// without miner/owner keypairs. We omit these details from the spec.
+// Also note that the pledge amount should be available in the ownerAddr in order for this call
+// to succeed.
 func (sms *StorageMiningSubsystem_I) CreateMiner(
+	state stateTree.StateTree,
 	ownerAddr addr.Address,
 	workerAddr addr.Address,
 	sectorSize util.UInt,
 	peerId libp2p.PeerID,
-) addr.Address {
-	// ownerAddr := sms.generateOwnerAddress(workerPubKey)
-	// var pledgeAmt actor.TokenAmount TODO: unclear how to pass the amount/pay
-	// TODO compute PledgeCollateral for 0 bytes
-	// return sms.StoragePowerActor().CreateStorageMiner(ownerAddr, workerPubKey, sectorSize, peerId)
-	// TODO: access this from runtime
-	// return sms.StoragePowerActor().CreateStorageMiner(ownerAddr, workerAddr, peerId)
-	var minerAddr addr.Address
-	return minerAddr
+	pledgeAmt actor.TokenAmount,
+) (addr.Address, error) {
+
+	params := make([]util.Serialization, 4)
+	params[0] = addr.Serialize_Address(ownerAddr)
+	params[1] = addr.Serialize_Address(workerAddr)
+	params[2] = libp2p.Serialize_PeerID(peerId)
+
+	sysActor, ok := state.GetActor(addr.SystemActorAddr)
+	Assert(ok)
+
+	unsignedCreationMessage := &msg.UnsignedMessage_I{
+		From_:       ownerAddr,
+		To_:         addr.StoragePowerActorAddr,
+		Method_:     ai.Method_StoragePowerActor_CreateStorageMiner,
+		Params_:     params,
+		CallSeqNum_: sysActor.CallSeqNum(),
+		Value_:      pledgeAmt,
+		GasPrice_:   0,
+		GasLimit_:   msg.GasAmount_SentinelUnlimited(),
+	}
+
+	var workerKey filcrypto.SigKeyPair // sms._keyStore().WorkerKey()
+	signedMessage, err := msg.Sign(unsignedCreationMessage, workerKey)
+	if err != nil {
+		return nil, err
+	}
+
+	err = sms.FilecoinNode().SubmitMessage(signedMessage)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: WAIT for block reception
+	util.IMPL_TODO()
```

Comment:
Choosing to leave these sorts of events where the subsystem waits for an appropriate state message before resuming execution as an IMPL_TODO given it will be very architecture dependent.

See https://github.com/filecoin-project/lotus/blob/a028f22ee687956a28341132bf84eff657710688/cmd/lotus-storage-miner/init.go#L566 for an example